### ISSUE TITLE
Squiz.NamingConventions.ValidVariableName how to only report for declaration, not usage

### ISSUE THREAD

**Issue by momala454** [squizlabs/PHP_CodeSniffer#3714]
**Describe the bug**
I want to report variable or members not in camelCase, but with some exceptions. So I would like to report only on variable/member declaration, not on usage

**Code sample**
```php
class Hello
{
    // phpcs:disable Squiz.NamingConventions.ValidVariableName
// it is ignored
    public string $hello_world;

    public function __construct(
// it is ignored
        public int $number_of_results,
    ) {
    }

    // phpcs:enable Squiz.NamingConventions.ValidVariableName

    public function test(): void
    {
        // shouldn't be reported
        $this->number_of_results = 1;

        // should be reported for the creation of $number_of_pages 
        $number_of_pages = $this->number_of_results;
    }
}
```

**Custom ruleset**
```xml
<?xml version="1.0"?>
<rule ref="Squiz.NamingConventions.ValidVariableName"/>
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```
 55 | ERROR | Member variable "number_of_results" is not in valid camel caps format
 58 | ERROR | Variable "number_of_pages" is not in valid camel caps format
 58 | ERROR | Member variable "number_of_results" is not in valid camel caps format
```

**Expected behavior**
An option to disable on usage (or disable if it has been disabled on the variable/member declaration). I don't want to need to write //phpcs::disable ... every time I use $this->number_of_results

**Versions (please complete the following information):**
 - OS: windows 10
 - PHP: 8.1
 - PHPCS: 3.7.1
 - Standard: psr12

**Additional context**

**Comment by gsherwood** [comment]
The sniff cannot be configured to only report on the first definition of a variable. It will always report everywhere the variable is used.

This isn't something I would add to the sniff as it is very hard to track this sort of variable declaration in PHP_CodeSniffer, so the results would not be accurate.

**Comment by momala454** [comment]
is it possible to detect if the variable is in left side of an equal (=) to differenciate assignment and usage ?

**Comment by runthis** [comment]
This would be nice. A better example would be from receiving an object from a vendor and there isn't a way.

```
$someVendor = new SomeVendor();
echo $someVendor->variable_isnt_mine;
```
